### PR TITLE
Implement version endpoint of storage api

### LINKED ISSUE #67: Implement Repository API version Endpoints
GET /apis/{api_id}/versions
GET /apis/{api_id}/versions/{version_id}
GET /apis/{api_id}/versions/{version_id}/definition

relates to #16

### PR DESCRIPTION
**Pull request by maschleg**
closes #67

### CODE REVIEW

File: storage/src/main/java/org/zalando/apidiscovery/storage/api/VersionListDto.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class VersionListDto {
+
+    private List<VersionsDto> versions;
```
Comment (maxim-tschumak) [inline storage/src/main/java/org/zalando/apidiscovery/storage/api/VersionListDto.java]: Minor: I would initialize versions with an empty list.

File: storage/src/main/java/org/zalando/apidiscovery/storage/api/ApiResourceController.java [hunk 2]
```diff
@@ -70,4 +92,27 @@ private ApiDto buildLinks(ApiDto api, UriComponentsBuilder builder) {
         return api;
     }
 
+    private List<VersionsDto> buildAndSetApplicationLinks(List<VersionsDto> versions, UriComponentsBuilder builder) {
+        versions.forEach(versionsDto -> buildAndSetApplicationLinks(versionsDto, builder));
+
+        return versions;
+    }
+
+    private VersionsDto buildAndSetApplicationLinks(VersionsDto version, UriComponentsBuilder builder) {
```
Comment (tkrop) [inline storage/src/main/java/org/zalando/apidiscovery/storage/api/ApiResourceController.java]: Should be called `updateVersionWithLinks`.

File: storage/src/main/java/org/zalando/apidiscovery/storage/api/ApiResourceController.java [hunk 3]
```diff
@@ -70,4 +92,27 @@ private ApiDto buildLinks(ApiDto api, UriComponentsBuilder builder) {
...
+        version.getDefinitions()
+            .forEach(apiDefinitionDto -> buildAndSetApplicationLinks(apiDefinitionDto, builder));
+
+        return version;
+    }
+
+    private ApiDefinitionDto buildAndSetApplicationLinks(ApiDefinitionDto apiDefinitionDto, UriComponentsBuilder builder) {
```
Comment (tkrop) [inline storage/src/main/java/org/zalando/apidiscovery/storage/api/ApiResourceController.java]: Should be called `updateApiDefinitionWithLinks`.

File: storage/src/main/java/org/zalando/apidiscovery/storage/api/ApiResourceController.java [hunk 4]
```diff
@@ -44,20 +44,42 @@ public ApiResourceController(ApiService apiService) {
     @GetMapping("/{api_id}")
     public ResponseEntity<ApiDto> getApi(@PathVariable("api_id") String apiId, UriComponentsBuilder builder) {
         return apiService.getApi(apiId)
-            .map(api -> ResponseEntity.ok(buildLinks(api, builder)))
+            .map(api -> ResponseEntity.ok(buildAndSetLinks(api, builder)))
```
Comment (tkrop) [inline storage/src/main/java/org/zalando/apidiscovery/storage/api/ApiResourceController.java]: This name change making it worse.

**Comment by maschleg** [comment]
@tkrop please review changes

**Comment by tkrop** [comment]
:+1:

**Pull request merged**